### PR TITLE
fix(core): the table indent reference edge depends on the document's compatibility mode

### PR DESCRIPTION
**Pull request by jan-kubica**
## What was wrong

`w:tblInd` does not measure from a fixed edge. The reference edge depends on the
document's `compatibilityMode`:

- from mode 15 on, the indent offsets the **table border**, and the leading
  cell margin pushes the cell text further in;
- in earlier modes, and in a document that declares no `compatSetting` at all,
  the indent measures to the **leading cell's text edge**: the border is pulled
  back by that cell's leading margin so the first character lands exactly on the
  indent.

The layout engine applied the mode-15 rule to every document, so in an older
document every table, and every line inside it, sat one cell margin too far in.
`w:tblCellMar` and `w:tcMar` change how far, so the error varied by document.

## How it is derived now

`resolveTableIndentCompatibility` reads the policy once from the document
settings and `toFlowBlocks` stamps it onto every table block, the way
justified-line fitting already carries its own generation policy. Body,
header, footer and footnote flows therefore agree without a second decision
point.

`resolveTableInlinePlacement` subtracts the leading cell's margin only for the
leading, indent-bearing edge. Centred and trailing-edge tables keep their
border-box placement in both modes, and RTL tables take the leading margin from
the trailing side of the first cell.

Two related defects fall out of putting the rule in one place:

- the header/footer painter re-derived the placement itself, so it read `w:jc`
  and `w:tblInd` directly, ignored `w:bidiVisual`, and could not see the new
  policy. It now calls `resolveTableInlineOffset`, leaving one owner for the
  rule.
- table conversion applies `w:tblInd` only when the measurement is `dxa`.
  Stamping the text-edge policy on a table whose indent was discarded left the
  compensation without its indent and moved the table into the margin by a cell
  margin nothing asked for. The policy now travels only with an indent the
  converter actually applied.

## Tests

- `layout-engine/tableIndentCompatibility.test.ts` covers the mode mapping,
  including the unset case.
- `layout-engine/measure/tableInlinePlacement.test.ts` adds a text-edge suite:
  leading-edge compensation for LTR and RTL, an unindented table, an explicit
  zero indent, and the untouched centred and trailing-edge cases.
- `layout-engine/tableRowBreak.test.ts` asserts the laid-out fragment geometry
  for both directions: the leading cell text lands on the authored indent.
- `layout-bridge/convert/toFlowBlocks-table-bidi.test.ts` covers policy
  threading, including the dropped-measurement case.
- `layout-painter/renderPage.test.ts` asserts an inline header table's position
  under both policies.

The existing border-edge tests are unchanged, which is the point: mode 15
documents lay out exactly as before.

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected table indentation and horizontal placement for documents using older compatibility modes.
  - Ensured tables in body content, headers, footers and footnotes follow consistent indentation rules.
  - Improved placement for left-to-right and right-to-left tables, including tables with cell padding.
  - Preserved existing positioning for newer document formats and supported alignment modes.

- **Tests**
  - Added coverage for legacy compatibility behaviour, unsupported indentation values, and inline tables in headers and footers.

### CODE REVIEW

File: packages/core/src/layout-bridge/convert/toFlowBlocks.ts [hunk 1]
```diff
@@ -2448,6 +2454,9 @@ function convertTable(node: PMNode, startPos: number, options: FlowConversionOpt
   if (indentPx !== undefined) {
     tableBlock.indent = indentPx;
   }
+  if (options.tableIndentCompatibility && !dropsAuthoredIndent) {
+    tableBlock.indentCompatibility = options.tableIndentCompatibility;
+  }
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/core/src/layout-bridge/convert/toFlowBlocks.ts:2459]: An absent `w:tblInd` is not "no indent" for this rule. Under the pre-2013
reference edge the leading border sits one leading cell margin outside the
content edge whether or not an indent is authored: the unindented case is
`0 - leadingCellMargin`, and the leading cell's text lands on the content edge
rather than one cell margin inside it. Gating on `indentPx !== undefined` would
restore the border-edge placement for every unindented table in an older
document, which is the larger half of the defect this PR fixes.
`resolveTableInlinePlacement with a text-edge indent > moves an unindented table
border out by the default leading cell margin` pins that case.

`dropsAuthoredIndent` is deliberately narrower: it fires only when the table
authors a `w:tblInd` whose measurement type the converter cannot apply, so the
indent is discarded. Keeping the policy there would apply the compensation
without the indent it belongs to. The absence of any `w:tblInd` is a different
state and keeps the policy.

File: packages/core/src/layout-engine/measure/tableInlinePlacement.ts [hunk 2]
```diff
@@ -1,24 +1,43 @@
...
     return { alignment: "center" };
   }
 
-  // w:tblInd adds space before the table's leading edge. Cell margins affect
-  // content inside that edge and therefore must not move the table itself.
   // OOXML justification is logical: bidiVisual mirrors left/right only after
   // deciding whether the leading-edge indent applies.
-  const offset = logicalJustification === "left" ? (table.indent ?? 0) : 0;
+  const offset =
+    logicalJustification === "left" ? (table.indent ?? 0) - leadingEdgeCompensation(table) : 0;
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/core/src/layout-engine/measure/tableInlinePlacement.ts:40]: An absent `w:tblInd` is not "no indent" for this rule. Under the pre-2013
reference edge the leading border sits one leading cell margin outside the
content edge whether or not an indent is authored: the unindented case is
`0 - leadingCellMargin`, and the leading cell's text lands on the content edge
rather than one cell margin inside it. Gating on `indentPx !== undefined` would
restore the border-edge placement for every unindented table in an older
document, which is the larger half of the defect this PR fixes.
`resolveTableInlinePlacement with a text-edge indent > moves an unindented table
border out by the default leading cell margin` pins that case.

`dropsAuthoredIndent` is deliberately narrower: it fires only when the table
authors a `w:tblInd` whose measurement type the converter cannot apply, so the
indent is discarded. Keeping the policy there would apply the compensation
without the indent it belongs to. The absence of any `w:tblInd` is a different
state and keeps the policy.
